mbbsd/bbs.c

```diff
@@ -3690,6 +3711,32 @@ mark_post(int ent, fileheader_t * fhdr, const char *direct)
     return change_post_mode(ent, fhdr, direct, FILE_MARKED);
 }
 
+static int
+mark_suspicious(int ent, fileheader_t * fhdr, const char *direct)
+{
+    boardheader_t *bp = getbcache(currbid);
+
+    aidu_t aidu = 0;
+    aidu = fn2aidu((char *)fhdr->filename);
+    if (aidu > 0) {
+        char aidc[10];
+        char buf[256];
+        aidu2aidc(aidc, aidu);
+
+        if (fhdr->filemode & FILE_SUSPICIOUS)
+            snprintf(buf, sizeof(buf),
+                     "%s �N #%s (%s) �Ѱ��ݬd��", cuser.userid, aidc, bp->brdname);
```

Comment:
`currboard` can be used instead of `getbcache(currbid)` if all you need is the board name.